File: backend/src/main/java/com/udyaa/rupiahpay/dto/ResponseAkun.java
package com.udyaa.rupiahpay.dto;

import java.util.Date;
import java.util.UUID;

import com.udyaa.rupiahpay.enums.AkunRole;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResponseAkun {
    private UUID uuid;
    private String firstName;
    private String middleName;
    private String lastName;
    private String email;
    private String password;
    private Date createdAt;
    private AkunRole role;
    private RestrictedResponseRekening balance;
}

File: frontend/src/interface/ResponseAkun.ts
import type { ResponseRekening } from "./ResponseRekening";

export interface ResponseAkun {
    uuid: string;
    firstName: string;
    middleName: string | undefined;
    lastName: string | undefined;
    email: string;
    password: string;
    createdAt: string;
    role: string;
    rekening: ResponseRekening
}